how-does-fix-with-copilot-work.md

```diff
@@ -0,0 +1,90 @@
+# How Does *Fix With Copilot* Work?
+In VS Code, Copilot has a feature called *Fix With Copilot* that offers to fix any error in the editor with AI. It's like a code fix for every error imaginable, with the caveat that Copilot is just trying its best.
+
+	Show an example here.
+	
+The Copilot extension intercepts every single error (diagnostic) that vscode gets from a language service
+(TODO: Omit or find out whether it's actually subscribed to every one or if there is a generic subscription).
+Then it builds a prompt that, at its core, says "There is an error with ***Error text here*** in this code ***Code here***. Please fix it."
+
+This works well for many errors, because the error implies a fix, and the fix is simple enough for Copilot to figure out. 
+
+## Targeted Fixes For Specific Errors
+Fix With Copilot fails when the context or the error are misleading. Most errors in Typescript and ESLint, which I work with day-to-day, actually try to avoid prescribing a fix in cases where it's not obvious. That's the opposite of what Copilot needs -- if the error isn't helpful, then all it has to use is the context of surrounding code. For example, ESLint has a great rule that detects duplicate if/else predicates:
+
+![Invoking AI](images/ai-no-dupe-if-1.png)
+
+The error message suggests that the duplicate is extraneous. Which it is, from a strictly control-flow perspective. But if we stop and look at *intent*, in my experience a duplicated `if/else` or `switch case` is the result of copy/paste/modifying cases until you get enough--and then forgetting to change the last one. That's why this ESLint rule is so great, because it catches attentional mistakes. Still, the fix is to make the duplicate case *different* from the others, not to delete.
```

Comment:
It's hard to imagine what you're trying to showcase since these images weren't included in the PR. Still, I'd play with the ordering here:


```suggestion
The error message suggests that the duplicate `if` check is extraneous. That's technically true... but what happens if we ask Copilot for a fix here?

[!Copilot suggesting deleting the `if` check entirely](...)

Copilot manages to get rid of the error -- *by deleting the entire block*. But if that was the right solution, we'd be better off avoiding coding errors by not writing any code (and just to check in, have you gone outside today?).

To get the best fix, we need to consider the intent -- *why* might the code be duplicated? How did the author get into this situation? And a little bit of empathy goes a long way here. I know when I've written a series of `if/else/if` checks, or `case`s in a `switch`, I tend to just copy/paste and modify the blocks -- but I also tend to forget to update that code with a new value to check against. And that's actually why this ESLint rule is so great: because it catches attentional mistakes.

So the best fix isn't to delete the code, it's to make the duplicate case *different* from the others.
```

Reply:
The rich diff shows the images for me, since they're already in the repo.

I re-ordered the presentation according to your high-level suggestion -- but didn't expand the text as much as this suggestion. Let me know if you think it needs more explanation here.